### PR TITLE
Resolve issues #460, #615, #625 by using js_regex (rebased).

### PR DESCRIPTION
**Pull request by janosch-x**

### CODE REVIEW

**Comment by tagliala** [comment]
Amazing :+1:

File: lib/client_side_validations/core_ext/regexp.rb [hunk 1]
```diff
@@ -1,18 +1,8 @@
 class Regexp
+  require 'js_regex'
```
Comment (tagliala) [inline lib/client_side_validations/core_ext/regexp.rb]: After merging, can I move this to `lib/client_side_validations.rb` ?

Comment (janosch-x) [inline lib/client_side_validations/core_ext/regexp.rb]: Maybe, but the current position is more robust.

If you move it, you *might* have to make sure that `client_side_validations.rb` is always loaded before `core_ext/regexp.rb` in production and before `test_core_ext.rb` in tests to protect against const_missing errors (depending on load orders and CI setup).

Comment (tagliala) [inline lib/client_side_validations/core_ext/regexp.rb]: moving it at the top of `lib/client_side_validations/core_ext/regexp.rb` (outside the `class Regexp`)?

Comment (janosch-x) [inline lib/client_side_validations/core_ext/regexp.rb]: Sure, if you prefer that!

**Comment by tagliala** [comment]
@jaynetics 

After upgrading the js_regex dependency from 3.4.0 to 3.5.0, this test is failing:

```rb
  def test_extended_mode_regexp_with_escaped_whitespace_as_json
    # regression test for issue #625
    test_regexp = /    [\ a]\    /x
    expected_regexp = { source: '[\\x20a]\\ ', options: '' }
    assert_equal expected_regexp, test_regexp.as_json
  end
```

Result:
```
Failure:
CoreExtTest#test_extended_mode_regexp_with_escaped_whitespace_as_json [/Users/geremia/dev/client_side_validations/test/core_ext/cases/test_core_ext.rb:91]:
--- expected
+++ actual
@@ -1 +1 @@
-{:source=>"[\\x20a]\\ ", :options=>""}
+{:source=>"[\\x20a] ", :options=>""}
```

I'm running `/    [\ a]\    /x` in irb and `/[\x20a] /` in js console, and they are working the same.

Should I change the test or there is a breaking change?

Ref: #625

**Comment by jaynetics** [comment]
@tagliala
you can change the test.

this is a cosmetic change as mentioned in js_regex' changelog.

`/\ /` and `/ /` have the same matching properties in js.

**Comment by tagliala** [comment]
@jaynetics thanks!

**Pull request merged**